api/python/provisioner/ui/form_window.py

```diff
@@ -0,0 +1,202 @@
+            result = result + f"{key}: {self.data[key]['default']} "
+        return result
+
+    def submit_button(self, x, y, menu, selected_rows):
+        buttons = {
+           "Submit": x + 15,
+           "Cancel": x + 5
+        }
+        selected_button = None
+
+        if selected_rows == len(menu):
+            selected_button = "Submit"
+        elif selected_rows == len(menu) + 1:
+            selected_button = "Cancel"
+
+        for button in buttons.keys():
+            if selected_button == button:
+                cod = ColorCode.get_color_pair(1)
+                self.on_attr(cod)
+                self._window.addstr(y + 1, buttons[button], button)
+                self.off_attr(cod)
+            else:
+                self._window.addstr(y + 1, buttons[button], button)
+
+    @staticmethod
+    def parse_input(kwargs):
+        if 'selected' not in kwargs:
+            selected_rows = 0
+        else:
+            selected_rows = kwargs['selected']
+
+        if 'edit' not in kwargs:
+            edit = False
+        else:
+            edit = kwargs['edit']
+        return selected_rows, edit
+
+    def action(self):
+        pass
+
+    def create_window(self, **kwargs):
+        color_code = kwargs['color_code']
+        self._parent = kwargs['component']
+
+        selected_rows, edit = FormWindow.parse_input(kwargs)
+        self._window.border()
+        col_code_attr = ColorCode.get_color_pair(color_code)
+        self.create_menu_head()
+
+        x = self.get_max_width() // 2
+        y = self.get_max_height() // 3 - 1
+        self.enable_keypad()
+
+        self.on_attr(col_code_attr)
+        self._window.addstr(y, 3, f"Please enter {self.component_type}"
+                            " for this machine")
+        self.off_attr(col_code_attr)
+
+        values = list(self.data.keys())
+        count_m = len(values)
+        x = 6
+
+        for idx, row in enumerate(values):
+            y = self.get_max_height() // 3 - count_m // 2 + (idx + 1) * 2
+            if not (idx == selected_rows):
+                self._window.addstr(y, x, f"{row}:")
+                self._window.addstr(y, 24, f" {self.data[row]['default']}")
+
+        y = self.get_max_height() // 3 - count_m // 2 + (count_m + 1) * 2
+
+        # Cancel and Submit button
+        self.submit_button(x, y, values, selected_rows)
+
+        if selected_rows < count_m:
+            y = (self.get_max_height() // 3 -
+                 count_m // 2 + (selected_rows + 1) * 2)
+
+            self.on_attr(col_code_attr)
+            self._window.addstr(y, x-3, ">> ")
+            self._window.addstr(y, x, f"{values[selected_rows]}:")
+
+            if edit:
+                if ('validation' in self.data[values[selected_rows]]):
+                    validate = self.data[values[selected_rows]]['validation']
+                else:
+                    validate = None
+                user_content = TextBox(
+                    self,
+                    1,
```

Comment:
hard-codes: please unwrap the magic

Reply:
This is length of textbox window, added this value in config fetching from config now.

Reply:
addressed